numpy/core/tests/test_function_base.py

```diff
@@ -362,3 +362,9 @@ def test_retstep(self):
                 assert_(isinstance(y, tuple) and len(y) == 2 and
                         len(y[0]) == num and isnan(y[1]),
                         'num={0}, endpoint={1}'.format(num, ept))
+
+    def test_object(self):
+        start = array(1, dtype='O')
+        stop = array(2, dtype='O')
+        y = linspace(start, stop, 3)
+        assert_array_equal(y, array([1., 1.5, 2.]))
```

Comment:
Might check that the result is an object array.

Reply:
Or should it be?

Reply:
I had that test first, but it failed...  And It should, as the `start = asanyarray(start) * 1.0` turns it into a regular float. And for floats, one would expect an array.

Note that I'm not sure this is necessarily the best behaviour, since it means that for very large ints, as in the original issue (#13089), we'll get back floats that are identical. But I don't know how to solve those anyway, since they cannot hold fractions. In the end, I think this is not numpy's problem... But we can at least avoid raising unrelated error on object input...

Reply:
OK, I hadn't checked, but was thinking of object arrays of high precision floats, mpmath or some such. Could be made to work, but not worth the trouble, those who need it can roll their own.

Reply:
Precision could be a problem even if the floats don't repeat because float precision decreases with amplitude. I remember having that problem with plot axis in single precision wavenumbers :) Maybe worth a note in the documentation at some point.